metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/force_delete_test.clj

```diff
@@ -66,3 +75,82 @@
        ;; tag association associated with other collection revision or whole collection is not deleted
        (util/is-tag-association-deleted? ta1 false)
        (util/is-tag-association-deleted? ta3 false)))))
+
+(deftest force-delete-service
+  (let [coll (util/create-and-save-collection "REG_PROV" 1)
+        coll-concept-id (:concept-id coll)
+        misc-svc (util/create-and-save-service "REG_PROV" 42)
+        misc-svc-assn (util/create-and-save-service-association
+                       coll misc-svc 1)
+        expected-svc-concept-id "S1200000003-REG_PROV"
+        expected-revision-ids [1 2 3]
+        revision-count 3]
+    (testing "delete most recent revision"
+      (let [latest-revision (util/create-and-save-service
+                             "REG_PROV" 1 revision-count)
+            concept-id (:concept-id latest-revision)
+            svc-assn (util/create-and-save-service-association
+                      coll latest-revision 1)]
+        (testing "initial conditions"
+          ;; creation results as expected
+          (is (= expected-svc-concept-id concept-id))
+          (is (= (last expected-revision-ids) (:revision-id latest-revision)))
+          ;; service revisions in db
+          (is (= 4
+                 (count (:concepts (util/find-concepts :service)))))
+          (is (= 200 (:status (util/get-concept-by-id-and-revision
+                               concept-id (last expected-revision-ids)))))
+          (is (= 200 (:status (util/get-concept-by-id-and-revision
+                               concept-id (second expected-revision-ids)))))
+          (is (= 200 (:status (util/get-concept-by-id-and-revision
+                               concept-id (first expected-revision-ids)))))
+          ;; make sure service association in place
+          (is (= "SA1200000004-CMR" (:concept-id svc-assn)))
+          (is (= coll-concept-id
+                 (get-in svc-assn [:extra-fields :associated-concept-id])))
+          (is (= expected-svc-concept-id
+                 (get-in svc-assn [:extra-fields :service-concept-id])))
+          ;; make sure collection associations in place
+          (is (not (:deleted (:concept (util/get-concept-by-id "SA1200000004-CMR"))))))
+        (testing "just the most recent revision is deleted"
+          (util/force-delete-concept concept-id (last expected-revision-ids))
+          (is (= 404 (:status (util/get-concept-by-id-and-revision
+                               concept-id (last expected-revision-ids)))))
+          (is (= 200 (:status (util/get-concept-by-id-and-revision
+                               concept-id (second expected-revision-ids)))))
+          (is (= 200 (:status (util/get-concept-by-id-and-revision
+                               concept-id (first expected-revision-ids))))))
+        (testing "service association has been deleted"
+          (is (:deleted (:concept (util/get-concept-by-id "SA1200000004-CMR")))))))
+    (testing "delete oldest revision"
```

Comment:
I don't think it is necessary to test deletion of the oldest revision. Just delete a middle revision is fine. And it would be better to move the deletion of revision 2 to above the deletion of the latest revision (revision 3), and show that revision 2 is deleted, but service association is not; then after delete revision 3, the service association is gone.

Reply:
Done.